client/export_test.go

```diff
@@ -26,19 +27,34 @@ var (
 )
 
 func (client *Client) SetDoer(d doer) {
-	client.doer = d
+	client.Requester().(*DefaultRequester).doer = d
 }
 
 func (client *Client) Do(method, path string, query url.Values, body io.Reader, v interface{}) error {
```

Comment:
Makes little sense to export this for tests here, now that we have a public requester. It's okay to postpone this for now, but it would be nice to clean it up as a follow up. Can we please leave just a TODO note here pointing that out?

Reply:
I added the following comment in export_test.go:

```
+// TODO: Clean up tests to use the new Requester API. Tests do not generate a client.response type
+// reply in the body while SyncRequest or AsyncRequest responses assume the JSON body can be
+// unmarshalled into client.response.
```